### PR TITLE
devhub: Add GITHUB_TOKEN

### PR DESCRIPTION
**Pull request by cb22**
Fixes https://github.com/tigerbeetle/tigerbeetle/actions/runs/10057090484/job/27797326947

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -147,6 +147,7 @@ jobs:
         env:
           DEVHUBDB_PAT: ${{ secrets.DEVHUBDB_PAT }}
           NYRKIO_TOKEN: ${{ secrets.NYRKIO_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (matklad) [inline .github/workflows/ci.yml:150]: Hm, shouldn't this literally be ` GH_TOKEN: ${{ github.token }}`, such that its a read-only token?

The `secrets.GITHUB_TOKEN` can public a new release, and that's not something we want to be able do from within devhub

Comment (cb22) [reply .github/workflows/ci.yml:150]: So `github.token` and `secrets.GITHUB_TOKEN` are [identical](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret) according to GitHub.

I think we should change `GITHUB_TOKEN` to be [default restrictive](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#setting-the-permissions-of-the-github_token-for-your-repository), and then add back permissions for publish.

Comment (matklad) [reply .github/workflows/ci.yml:150]: Oh, excellent catch, thanks!

**Review by matklad — APPROVED** [review]

**Comment by cb22** [comment]
I've changed the GitHub permissions setting:
image

**Pull request merged**